modules/storages/app/models/storages/upload_link.rb

```diff
@@ -0,0 +1,8 @@
+class Storages::UploadLink
+  attr_reader :destination, :finalize
+
+  def initialize(destination = URI(''), finalize = URI(''))
```

Comment:
what's the purpose of using empty uris instead of `nil` for the absence of data?

Reply:
As I modeled it with `URI` I wanted to show the programmer what to pass in here. I could switch to string but then it would still be the empty string as default. Better would be either an Optional/Maybe or making the constructor private and provide a class method that validated the type that is passed in.

What would you choose?

Reply:
Like that e.g.:

```
class Storages::UploadLink
  attr_reader :destination, :finalize

  def self.create(destination, finalize)
    new(URI::parse(destination), URI::parse(finalize))
  end

  private

  def initialize(destination = '', finalize = '')
    @destination = destination
    @finalize = finalize
  end
end
```